metau-capstone/API/APIManager.m

```diff
@@ -34,16 +35,19 @@ - (instancetype)init {
     return self;
 }
 
-- (void) getSheetsData: (void(^)(NSError *error))completion {
+- (void) getSheetsData: (NSString *) pathParameters withCompletetion: (void(^)(NSError *error))completion {
     NSDictionary *parameters = @{@"key": self.APIkey};
     AFHTTPSessionManager *manager = [[AFHTTPSessionManager alloc] initWithBaseURL:self.baseURL];
     manager.requestSerializer = [AFJSONRequestSerializer serializer];
-    
-    [manager GET:@"v4/spreadsheets.json" parameters:parameters headers:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, NSDictionary *  _Nullable sheetDictionary) {
+    NSString *endURLString = [@"v4/spreadsheets/" stringByAppendingString:pathParameters];
+    [manager GET:endURLString parameters:parameters headers:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, NSDictionary *  _Nullable sheetDictionary) {
         // Success
         completion(nil);
+        [Flashcard createCardsFromDictionary:sheetDictionary];
```

Comment:
nit: often a good practice in software development is single responsibility (S from SOLID principle). when applying in this case, this file is API manager and the method is getSheetsData, it would best to just handle the fetch of sheets data and return the payload to the caller via completion handler. the handler then can invoke `[Flashcard createCardsFromDictionary:sheetDictionary];` or whatever it desires

Reply:
done!